CoreMentoringApp.WebSite.Tests/Areas/Api/Controllers/CategoriesControllerTests.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using AutoMapper;
+using CoreMentoringApp.Core.Models;
+using CoreMentoringApp.Data;
+using CoreMentoringApp.WebSite.Areas.Api.Controllers;
+using CoreMentoringApp.WebSite.Areas.Api.Models;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using Xunit;
+
+namespace CoreMentoringApp.WebSite.Tests.Areas.Api.Controllers
+{
+    public class CategoriesControllerTests
+    {
+
+        private readonly Mock<IDataRepository> _mockDataRepository;
+        
+        private readonly Mock<IMapper> _mockMapper;
+
+        public CategoriesControllerTests()
+        {
+            _mockDataRepository = new Mock<IDataRepository>();
+            _mockMapper = new Mock<IMapper>();
+        }
+
+        [Fact]
+        public void Get_ReturnsListOfCategoriesDTO()
+        {
+            var testCategories = GetTestCategories();
+            var testCategoriesDto = GetTestCategoriesDTO();
+            _mockDataRepository.Setup(repo => repo.GetCategories())
+                .Returns(testCategories)
+                .Verifiable();
+            _mockMapper.Setup(m => m.Map<IEnumerable<CategoryDTO>>(It.IsAny<IEnumerable<Category>>()))
+                .Returns(testCategoriesDto)
+                .Verifiable();
+            var controller = new CategoriesController(_mockDataRepository.Object, _mockMapper.Object);
+
+            var result = controller.Get();
```

Comment:
Completely not an issue, just wanted to share my opinion about such tests for controllers. For me, such tests are just a waste of time. It's a unit test for the controller, but controllers should be thin as possible and should not contain complicated logic - your controllers is a good example. 
In real-world client is not going just call controller.Get(). Instead, the request will go through the pipeline (routing, binding e.t.c) and that is what is more interesting to test. This type of tests is called integration tests in .Net Core world. You still mock all dependencies of the controller, but using in-memory web-server and httpClient you invoke your controller actions using HTTP requests. This way you can tests that you routing, bindings, authorisation works as you expect.
Here is a link for your reference: https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-5.0

Reply:
Really nice point, even writing these unit tests was really boring. As I understand they could be useful only in case of possible future controller rework, to "not forget" ask repository about data, if it's possible. 
I've added several integration tests. Not sure that test for creation and getting created product it's the best example of AAA pattern, just wanted to test the complete process.

Reply:
Here is a PR with applied changes 
https://github.com/Siarhei-Sialitski/CoreMentoringApp/pull/5